src/compiler/sys.ts

```diff
@@ -500,7 +500,7 @@ namespace ts {
                 },
                 fileExists,
                 directoryExists,
-                createDirectory(directoryName: string) {
+                createDirectory(this: System, directoryName: string) {
                     if (!this.directoryExists(directoryName)) {

```

Comment:
Again, it'd consider if it may be better to just replace `this` with `sys` and not need a `this` type, since it also makes it so the function doesn't need to be rebound to be used.